CrossPlatformDesktopProject/CrossPlatformDesktopProject/Link/Player.cs

```diff
@@ -130,7 +137,13 @@ public void DrawSprite(SpriteBatch spriteBatch, Texture2D texture, Rectangle sou
             Rectangle destination = new Rectangle(
                 (int)xPos + XOffset, (int)yPos + YOffset,
                 source.Width * 3, source.Height * 3);
+            hitbox = destination;
             spriteBatch.Draw(texture, destination, source, Color.White);
         }
+
+        public Rectangle GetRectangle()
```

Comment:
Why couldn't this implementation be something like 

```C#
return new Rectangle(...).
```

Then there wouldn't need to be an extra field in the class. It seems to me like there should be one "source of truth" for the position: xPos and yPos, and then GetRectangle can construct on the fly the corresponding rectangle.

If anything, I think Draw() should reference GetRectangle(). It seems strange to me to have a Draw() method modify the object's internal state.

Reply:
For mos tof the methods I only implemented updates to the hitbox in the draw method because it was convenient. Each class should have updates to the hitbox be managed in whichever commands would be called soonest in the update order. 
Additionally, storing a rectangle in the class makes it easier to handle the changing sprite sizes, at least in the cases that that occurs. Otherwise we would need some other field to store the width and height of the rectanlge to be used by the new rectangle function